nav2_gps_waypoint_follower_demo/nav2_gps_waypoint_follower_demo/utils/gps_utils.py

```diff
@@ -0,0 +1,31 @@
+import math
+from geographic_msgs.msg import GeoPose
+from geometry_msgs.msg import Quaternion
+
+def quaternion_from_euler(roll, pitch, yaw):
```

Comment:
I'm 99.9% sure TF2 does this or numpy. Lets not reinvent what exists :-)

Reply:
I spent some time looking for an alternative built into ROS2 but wasn't able to find any. Seems that this functionality was removed from tf2 as [this](https://answers.ros.org/question/355794/ros2-python-quaternion-to-euler/) and [this](https://github.com/ros/geometry2/issues/222) suggests. 

People seem to point out to the [tf_transformations](https://index.ros.org/p/tf_transformations/) package, but, is it really worth it making users install the library in the tutorial only for this one function?

Reply:
Maybe not. But what about numpy or some standard python tool? Less math for us to potentially screw up, the better. This looks right to me, but its about future additions / changes

Reply:
I can't seem to find any builtin function to do this with python or numpy (at least not a function like `quaternion_from_euler`). alternatives are present in other libraries like 
- the already mentioned [tf_transformations](https://index.ros.org/p/tf_transformations/)
- [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html)
- various other packages that can be installed with pip

I would prefer not to install any of those and just implement this single function. Since we only care about the yaw angle I can change it to `quaternion_from_yaw`, which should reduce the math significantly. What do you think?

Reply:
OK, we can just keep it, not worth overoptimizing for tutorial code :-)